File: solutions/python/sum-of-two-values.py
def find_sum_of_two(A, val):
  found_values = set()
  for a in A:
    if val - a in found_values:
      return True
    found_values.add(a)
  return False

v = [5, 7, 1, 2, 8, 4, 3]
test = [3, 20, 1, 2, 7]

for i in range(len(test)):
  output = find_sum_of_two(v, test[i])
  print("find_sum_of_two(v, " + str(test[i]) + ") = " + str(output))

File: solutions/javascript/sum-of-two-values.js
let findSumOfTwo = function(A, val) {
  let foundValues = new Set();
  for (let a in A) {
    if (foundValues.has(val - A[a])) {
      return true;
    }
    foundValues.add(A[a]);
  }
  return false;
};

let v = [5, 7, 1, 2, 8, 4, 3];
let test = [3, 20, 1, 2, 7];

for (i = 0; i < test.length; i++) {
  let output = findSumOfTwo(v, test[i]);
  console.log("findSumOfTwo(v, " + test[i] + ") = " + output);
}
